DigitalOffice/Services/TimeManagementService/Validators/CreateWorkTimeRequestValidator.cs

```diff
@@ -0,0 +1,80 @@
+﻿using FluentValidation;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using TimeManagementService.Models;
+using TimeManagementService.Repositories.Filters;
+using TimeManagementService.Repositories.Interfaces;
+
+namespace TimeManagementService.Validators
+{
+    public class CreateWorkTimeRequestValidator : AbstractValidator<CreateWorkTimeRequest>
+    {
+        private IWorkTimeRepository repository;
+
+        // How many days ago can WorkTime be filled.
+        private readonly static int fromDay = 3;
+        private readonly DateTime fromDateTime = DateTime.Now.AddDays(-fromDay);
+        // How many days ahead can WorkTime be filled.
+        private readonly static int toDay = 2;
+        private readonly DateTime toDateTime = DateTime.Now.AddDays(toDay);
+
+        public CreateWorkTimeRequestValidator([FromServices] IWorkTimeRepository repository)
+        {
+            this.repository = repository;
+
+            RuleFor(wt => wt.WorkerUserId)
+                    .NotEmpty();
+
+            RuleFor(wt => wt.Date)
+                .NotEmpty()
+                .When(wt => wt.Date > fromDateTime)
+                .WithMessage(wt => string.Format("WorkTime had to be filled no later than {0}.",
+                    fromDateTime.ToString(System.Globalization.CultureInfo.CreateSpecificCulture("en-GB"))))
+                .When(wt => wt.Date < toDateTime)
+                .WithMessage(wt => string.Format("WorkTime cannot be filled until {0}.",
+                    fromDateTime.ToString(System.Globalization.CultureInfo.CreateSpecificCulture("en-GB"))));
+
+            RuleFor(wt => wt.Time)
+                .NotEmpty()
+                .When(wt => TimeAndDateOfNewWorkTimeIsValid(wt.WorkerUserId, wt.Date, wt.Time));
+
+            RuleFor(wt => wt.ProjectId)
+                .NotEmpty();
+        }
+
+        private bool TimeAndDateOfNewWorkTimeIsValid(
```

Comment:
This should not be a separate method, because it will not produce specialized error messages, I will redo.

Reply:
And it would be nice to add comments.

Reply:
Not whole numbers are compared. Apparently you need to use a interval.

Reply:
Fixed.